MyPlayground.playground/Contents.swift

```diff
@@ -0,0 +1,85 @@
+func mean(_ numbers: [Int]) -> Double? {
+    if numbers.isEmpty { return nil }
+    
+    var sumOfNumbers = 0
+    
+    for number in numbers {
+        sumOfNumbers += number
+    }
+
+    let mean = Double(sumOfNumbers) / Double(numbers.count)
+
+    return mean
+}
+
+
+func median(_ numbers: [Int]) -> Double? {
+    
+    if numbers.isEmpty { return nil }
+    
+    let sortedNumbers = numbers.sorted()
+    let middleIndex = sortedNumbers.count / 2
+    
+    if sortedNumbers.count.isMultiple(of: 2) {
+        let median = Double(sortedNumbers[middleIndex] + sortedNumbers[middleIndex - 1]) / 2
+        return median
+    } else {
+        let median = Double(sortedNumbers[middleIndex])
+        return median
+    }
+}
+
+
+func modes(_ numbers: [Int]) -> [Int] {
+    
+    if numbers.isEmpty { return [] }
+
+    var counter = [Int: Int]()
+    var modes = Set<Int>()
+
+    for ( _ , value) in numbers.enumerated() {
+        if counter[value] != nil {
+            counter[value]! += 1
+        } else {
+            counter[value] = 1
+        }
+    }
+
+    let sortedCounters = counter.sorted { return $0.value > $1.value }
+    var maxValue = 1
+
+    for (key, value) in sortedCounters {
+        if maxValue > value {
+            break
+        } else {
+            modes.insert(key)
+            maxValue = value
+        }
+    }
+    
+    return modes.sorted()
+}
+
+
+let testNumbers = [1,2,3,8,2,3,2,4,4,4]
+
+
+if let mean = mean(testNumbers) {
+    print("Mean: \(mean)")
+} else {
+    print("Mean: not value")
```

Comment:
nit:
You can be more friendly to the user:
```suggestion
    print("There is no mean value in empty array.")
```

Reply:
Fixed.